public/components/trace_analytics/requests/traces_request_handler.ts

```diff
@@ -157,8 +158,37 @@ export const handleTracesRequest = async (
         })
       : Promise.resolve({});
 
-  return Promise.allSettled([responsePromise, percentileRangesPromise])
-    .then(([responseResult, percentileRangesResult]) => {
+  let rootSpansTotalHitsPromise: Promise<number> | undefined;
+
+  if (setTotalHits && (mode === 'data_prepper' || mode === 'custom_data_prepper')) {
+    const rootSpansDSL = cloneDeep(timeFilterDSL);
```

Comment:
Question: Why do we need cloneDeep? I might have missed something here.

Reply:
Clone deep is needed for the query to prevent any possible mutation as we adjust it to account for parentSpanId being empty.